python/tvm/relay/backend/contrib/ethosu/legalize.py

```diff
@@ -1242,6 +1242,114 @@ def __call__(self, *args, **kwargs):
         pass
 
 
+class Resize2dRewriter(DFPatternCallback):
+    """
+    Convert ethos-u.resize2d composite function to an equivalent operation that
+    performs the relevant upsampling operation.
+
+    Case 1: Upsample factor of 1 i.e. no upsampling:
+        add scalar of 0 - do nothing.
```

Comment:
Looks like you are doing an identity pooling instead of adding 0?

Reply:
Thanks that's correct, it was that before I changed my mind :) In the latest revision I removed support for "no up-sampling" entirely as I don't think it makes much sense, WDYT?

Reply:
Updated, thanks!